nix/overlay.nix

```diff
@@ -5,7 +5,7 @@ final: prev: {
   devShell = final.callPackage ./waypoint.nix { };
 
   # Need to manually do this since 1.16 is still the default
-  go = final.go_1_17;
+  go = final.go_1_18;
```

Comment:
@mitchellh Let me know if this isn't correct

Reply:
Probably not, because we pin the index so 1.18 is prob not available. Just leave it out and I'll upgrade myself later. :)